x/concentrated-liquidity/pool_store.go

```diff
@@ -1,22 +1,46 @@
 package concentrated_liquidity
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/osmosis-labs/osmosis/v12/osmoutils"
 	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
 )
 
-func (k Keeper) getPoolbyId(ctx sdk.Context, poolId uint64) Pool {
+func (k Keeper) MarshalPool(pool types.PoolI) ([]byte, error) {
+	return k.cdc.MarshalInterface(pool)
+}
+
+func (k Keeper) UnmarshalPool(bz []byte) (types.PoolI, error) {
```

Comment:
ditto

Also, having a marshal - unmarshal unit test would be great

Reply:
What do you think's a good test for the Marshal and unmarshalling? I don't think they do much + hard to test since we get returned into bytes. 

A test case I can think of Marshalling a pool and then unmarshalling it to see if we get the same result. wdyt?